src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> oldNode = tail;
+        Node<T> newNode = new Node<>(oldNode,value,null);
+        tail = newNode;
+        if (oldNode != null) {
+            oldNode.next = newNode;
+        } else {
+            head = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " don't exist!");
+        }
```

Comment:
You already have `checkIndex` method, you can pass `size` or `size + 1` in different cases as parameter to avoid code duplications.